### PR TITLE
Polish up docstrings and more

### PR DESCRIPTION
**Pull request by visr**
Sorry if this is too much of a mixed bag, would be happy to split it up as well. But it's mostly straightforward.

88d9964ca78a18d2c42e9a134184a588dbaee81e removes empty files
d6aca842107ff2a4fedc865a28daa31c2425a7d0 only touches docstrings
4cd00516c18c1cf2d2953e4e7cdee9d1f190d94c adds argument type annotations, good to double check if all are correct
82c15067199640f9895994fea3e8b8c538cdd89e is a simple do block
a4c409b7737c088250789ec228eb292c12f1e9d2 DArray.concat must be a function right?
d8f4fdee5b10a3cb58e03db2dacfd74dd88b10a2 looks like an oversight

### CODE REVIEW

File: src/compute.jl [hunk 1]
```diff
@@ -84,10 +91,12 @@ function dependents(node::Thunk, deps=Dict{Thunk, Set{Thunk}}())
 end
 
 """
-recursively find the number of taks dependent on each task in the DAG.
-Input: dependents dict
+    noffspring(dpents::Dict{Thunk, Set{Thunk}}) -> Dict{Thunk, Int}
+
+Recursively find the number of taks dependent on each task in the DAG.
```
Comment (jpsamaroo) [inline src/compute.jl]: Typo (from original source): `taks` -> `tasks`

File: src/fault-handler.jl [hunk 2]
```diff
@@ -1,4 +1,8 @@
-"Recursively checks if an exception was caused by a worker exiting."
+"""
+    check_exited_exception(res) -> Bool
```
Comment (jpsamaroo) [inline src/fault-handler.jl]: This is technically an internal function only used by the fault handler, but I think this is still a fine change. Maybe just add `res::Exception` to the argument to indicate that it takes an `Exception` as input?

File: src/processor.jl [hunk 3]
```diff
@@ -3,31 +3,45 @@ export OSProc, Context
...
+OS process - contains pid returned by `Distributed.workers`
 """
 struct OSProc <: Processor
     pid::Int
 end
 
 """
-A context represents a set of processors to use
-for a papply operation.
+A context represents a set of processors to use for a operation.
```
Comment (jpsamaroo) [inline src/processor.jl]: `a operation` -> `an operation`

File: src/processor.jl [hunk 4]
```diff
@@ -3,31 +3,45 @@ export OSProc, Context
...
+    Context(xs::Vector{OSProc}) -> Context
+    Context(xs::Vector{Int}) -> Context
+
+Create a Context, by default adding each available worker once from
+every available thread. Use the `nthreads` keyword to use a different
+number of threads.
+
+It is also possible to create a Context from a vector of [`OSProc`](@ref),
+or equivalently the underlying process ids can also be passed directly
+as a Vector{Int}.
```
Comment (jpsamaroo) [inline src/processor.jl]: Should `Vector{Int}` be between backticks?

**Comment by visr** [comment]
Thanks for having a look! I addressed the comments.

File: src/array/darray.jl [hunk 5]
```diff
@@ -237,7 +237,7 @@ end
 A DArray object may contain a thunk in it, in which case
 we first turn it into a Thunk object and then compute it.
 """
-function compute(ctx, x::DArray; persist=true, options=nothing)
+function compute(ctx::Context, x::DArray; persist=true, options=nothing)
```
[the remark this answers is not in the document]
Comment (visr) [reply src/array/darray.jl]: I cannot answer this, but just want to mention that a part of the signatures already had `::Context`. So probably we should either remove them all or add them all.

[the remark this answers is not in the document]
Comment (jpsamaroo) [reply src/array/darray.jl]: Yeah I agree with you, I guess we can just go with your changes, and then widen things as needed later on.

**Comment by jpsamaroo** [comment]
Thanks a ton!

**Pull request merged**